### DISCUSSION TITLE
Why does pipeline task with "depends" on, not wait before first step is done?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [vercel/turborepo#1144] [Help] [answered]
I don't understand why I in the following output see logs from `web-app:build` before `build-setup:setup` is done?

Did I misunderstand something here?

![Screen Shot 2022-04-28 at 12 29 55](https://user-images.githubusercontent.com/9967422/165733840-1cf62a46-2ce1-4e1a-8b27-5fe3a7e1274f.png)

I'm also unsure why I get this graph image - should `setup` not be the first thing?

![graph-1651142024091607000](https://user-images.githubusercontent.com/9967422/165734245-61d607c7-3cf4-4673-b032-8e119988c7bf.jpg)

**Comment by mehulkar** [comment, marked as answer]
If `build-setup` is a dependency of `web-app`, you should change your `dependsOn` to `["^setup", "^build"]`. The `^` denotes, that you want the those tasks from _dependencies_ to run and complete first.

**Reply by traviswimer** [reply]
What is the reasoning behind not waiting for the task to complete when not using `^`?

If I need a task that's not a dependency to run before another task, is the only/best option to use the specific package syntax (`package#task`)?
